sharktank/sharktank/ops/sharded_impls.py

```diff
@@ -669,6 +669,19 @@ def index_copy__split_replicated_split(
     return inout
 
 
+@index_put_.override(AllOfType(ReplicatedTensor))
+def index_put__replicated(
+    inout: ReplicatedTensor,
+    indices: Tuple[ReplicatedTensor],
```

Comment:
I am not sure it would look into the tuple when dispatching to see that all tensors are replicated.

Reply:
Actually the trampoline would take the tensors from the tuple.